src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
+                                string cwd = null;
+                                var pscwd = Context.EngineSessionState.InvokeProvider.Item.Get(".");
+                                Diagnostics.Assert(pscwd != null, "pscwd should never be null");
+                                if (pscwd.Count == 1)
+                                {
+                                    cwdinfo = pscwd[0].BaseObject as System.IO.DirectoryInfo;
+                                }
+                                if (cwdinfo != null)
+                                {
+                                    cwd = cwdinfo.FullName;
+                                }
+                                // If it's a filesystem location then expand the wildcards
+                                if (cwd != null)
+                                {
+                                    // See if there are any matching paths otherwise just add the pattern as the argument
+                                    var paths = Context.EngineSessionState.InvokeProvider.ChildItem.Get(arg, false);
+                                    if (paths.Count > 0)
+                                    {
+                                        bool first = true;
+                                        foreach (var path in paths)
+                                        {
+                                            object pbo = path.BaseObject;
+                                            if (! first)
+                                            {
+                                                _arguments.Append(" ");
+                                            }
+                                            else
+                                            {
+                                                if (! (pbo is System.IO.FileInfo || pbo is System.IO.DirectoryInfo))
+                                                {
+                                                    // If the object is not a filesystem object, then just append
+                                                    // the pattern unchanged
+                                                    _arguments.Append(arg);
+                                                    break;
+                                                }
+                                                first = false;
+                                            }
+                                            System.IO.FileInfo fi = pbo as System.IO.FileInfo;
+                                            if (fi != null)
+                                            {
+                                                _arguments.Append(computeRelativePath(cwd, fi.FullName));
+                                            }
+                                            else
+                                            {
+                                                System.IO.DirectoryInfo di = pbo as System.IO.DirectoryInfo;
```

Comment:
It's recommended to use [`is-expression`](https://blogs.msdn.microsoft.com/dotnet/2017/03/09/new-features-in-c-7-0/) introduced in C# 7 to replace the `as` operator because it can help reduce nested brackets. For example, here it can be:
```
if (pbo is FileInfo fi)
{
    ...
}
else if (pbo is DirectoryInfo di)
{
    ...
}
```